src/feature.rs

```diff
@@ -66,27 +67,28 @@ impl TryFrom<JsonObject> for Feature {
     type Error = Error;
 
     fn try_from(mut object: JsonObject) -> Result<Self, Error> {
-        match &*util::expect_type(&mut object)? {
+        let res = &*util::expect_type(&mut object)?;
+        match res {
             "Feature" => Ok(Feature {
                 geometry: util::get_geometry(&mut object)?,
                 properties: util::get_properties(&mut object)?,
                 id: util::get_id(&mut object)?,
                 bbox: util::get_bbox(&mut object)?,
                 foreign_members: util::get_foreign_members(object)?,
             }),
-            _ => Err(Error::GeoJsonUnknownType),
+            _ => Err(Error::NotAFeature(res.to_string())),
```

Comment:
For users handling specific errors, most of the breaking API changes will be straight forward:

e.g. diffs like
```diff
-     Err(Error::GeoJsonExpectedObject)
+     Err(Error::GeoJsonExpectedObject(value))
```

But might be worth including some migration details for the less obvious ones like this one:

`GeoJsonUnknownType` becomes either `NotAFeature` or `EmptyType`

Take it or leave it though - probably most users aren't handling specific errors.

Reply:
Good plan, will do.